file/file_prefetch_buffer.cc

```diff
@@ -275,15 +245,52 @@ Status FilePrefetchBuffer::PrefetchAsync(const IOOptions& opts,
     curr_ = curr_ ^ 1;
     second = curr_ ^ 1;
   }
+
   // After swap check if all the requested bytes are in curr_, it will go for
   // async prefetching only.
   if (bufs_[curr_].buffer_.CurrentSize() > 0 &&
       offset + length <=
           bufs_[curr_].offset_ + bufs_[curr_].buffer_.CurrentSize()) {
     offset += length;
```

Comment:
This is confusing. Some comments would help here. We're modifying offset and length, but how does the caller know that data has been found, and in which buffer? Even if caller wants to copy, the original offset and length have been modified.

Reply:
Yes, your understanding is right. I will change the name of the function and add more comments. Also I don't think I need to change offset and length. Caller can do it itself. I will fix it.

Reply:
Currently I am fixing some issues and making sure the behavior is as expected with seek and async_io. I will upload the changes once those issues are fixed.